labrad/wrappers.py

```diff
@@ -390,18 +390,76 @@ def _send(self, *args, **kw):
         return self._cxn._send(self.ID, *args, **kw)
 
 @inlineCallbacks
-def getConnection(host=C.MANAGER_HOST, port=C.MANAGER_PORT, name="Python Client", password=None):
+def getConnection(host=C.MANAGER_HOST, port=None, name="Python Client",
+                  password=None, tls=C.MANAGER_TLS):
     """Connect to LabRAD and return a deferred that fires the protocol object."""
-    p = yield protocol.factory.connectTCP(host, port, C.TIMEOUT)
+    tls = C.check_tls_mode(tls)
+    if port is None:
+        port = C.MANAGER_PORT_TLS if tls == 'on' else C.MANAGER_PORT
+
+    if tls == 'on':
+        tls_options = crypto.tls_options(host)
+        p = yield protocol.factory.connectSSL(host, port, tls_options,
+                                              timeout=C.TIMEOUT)
+    else:
+        def connect():
+            return protocol.factory.connectTCP(host, port, timeout=C.TIMEOUT)
+
+        @inlineCallbacks
+        def start_tls(p, cert_string=None):
+            try:
+                resp = yield p.sendRequest(C.MANAGER_ID, [(1L, ('STARTTLS', host))])
+            except Exception, e:
+                raise Exception(
+                    'Failed sending STARTTLS command to server. You should '
+                    'update the manager and configure it to support encryption '
+                    'or else disable encryption for clients. See '
+                    'https://github.com/labrad/pylabrad/blob/master/CONFIG.md')
+            cert = resp[0][1]
+            p.transport.startTLS(crypto.tls_options(host, cert_string=cert_string))
+            returnValue(cert)
+
+        def ping(p):
+            return p.sendRequest(C.MANAGER_ID, [(2L, 'PING')])
+
+        p = yield connect()
+        if ((tls == 'starttls-force') or
+            (tls == 'starttls' and p.transport.getPeer().host != '127.0.0.1')):
```

Comment:
The correct way to test for localhost is p.transport.getPeer().host in ["127.0.0.1", "::1"] or (p.transport.getPeer().host == p.transport.getHost().host)  This works for ipv6, and also works when a user on the local host connects by name rather than explicitly using localhost.